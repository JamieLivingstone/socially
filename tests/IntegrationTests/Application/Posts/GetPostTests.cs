using System.Linq;
using System.Threading.Tasks;
using Application.Common.Exceptions;
using Application.Posts.Queries.GetPost;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Posts;

[TestFixture]
public class GetPostTests : TestBase
{
  [Test]
  public void GivenPostDoesNotExist_ThrowsNotFoundException()
  {
    var query = new GetPostQuery
    {
      Slug = "does-not-exist",
    };

    async Task Handler() => await SendAsync(query);

    Assert.ThrowsAsync(typeof(NotFoundException), Handler);
  }

  [Test]
  public async Task GivenPostExists_ReturnsPost()
  {
    var query = new GetPostQuery
    {
      Slug = Seed.Posts().First().Slug,
    };

    var post = await SendAsync(query);

    Snapshot.Match(post);
  }
}
