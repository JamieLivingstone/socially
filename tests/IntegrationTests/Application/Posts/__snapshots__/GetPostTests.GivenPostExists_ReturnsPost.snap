{
  "Slug": "test-post-one",
  "Title": "Test Post One",
  "Body": "Test Post One Body",
  "CreatedAt": "2021-01-01T00:00:00",
  "UpdatedAt": "2021-01-01T00:00:00",
  "Author": {
    "Username": "liam"
  },
  "Tags": [],
  "CommentsCount": 4,
  "LikesCount": 0
}
